### PR TITLE
Checkout v2: Improve truncation of displayed addresses

### LINKED ISSUE #4919: [Bug]: Checkout V2 addresses are cut off
### What is your BTCPay version?

BTCPay Server v1.9.1+d738f797e

### How did you deploy BTCPay Server?

docker

### What happened?

When using a checkout V2 addresses both on chain and bolt 11 are truncated in a way it doesn't really allow for easy way to verify it.

Screenshot 2023-04-26 at 12 21 06

### How did you encounter this bug?

When using checkout v2
1. Create an invoice
2. Go to checkout

### Relevant log output

_No response_

### What browser do you use?

Safari Version 16.3 (18614.4.6.1.6)

### Additional information

I propose we do something like this instead 
Screenshot 2023-04-26 at 12 15 37

### Are you sure this is a bug report?

- [X] I confirm this is a bug report

### PR DESCRIPTION
**Pull request by dennisreimann**
Fixes #4919. Had to extend the component quite a bit to handle the responsive + JavaScript value cases, but it works well now :)

grafik

### CODE REVIEW

File: BTCPayServer/Components/TruncateCenter/Default.cshtml [hunk 1]
```diff
@@ -1,10 +1,29 @@
...
+        <span class="truncate-center-truncated" @(!string.IsNullOrEmpty(Model.Start) ? $"data-bs-toggle=tooltip title={Model.Text}" : "")>
+            <span class="truncate-center-start">@(Model.Elastic ? Model.Text : $"{Model.Start}…")</span>
+            <span class="truncate-center-end">@Model.End</span>
+        </span>
+        <span class="truncate-center-text">@Model.Text</span>
+    }
     @if (Model.Copy)
     {
-        <button type="button" class="btn btn-link p-0" data-clipboard="@Model.Text">
+        <button type="button" class="btn btn-link p-0" @(Model.IsVue ? ":" : string.Empty)data-clipboard=@Safe.Json(Model.Text)>
```
Comment (Kukks) [inline BTCPayServer/Components/TruncateCenter/Default.cshtml:26]: oof :D

**Review by Kukks — APPROVED** [review]
ACK in spirit but the design code is over my head

File: BTCPayServer/Components/TruncateCenter/Default.cshtml (hunk 1 above)
Comment (dennisreimann) [reply BTCPayServer/Components/TruncateCenter/Default.cshtml:26]: It gets the job done :D

**Comment by dennisreimann** [comment]
> the design code is over my head

The CSS is actually the cleanest part :D

**Review by pavlenex — APPROVED** [review]
tACK

**Review by dstrukt — APPROVED** [review]
tACK - LGTM!

**Pull request merged**